pkg/webhook/channel_validator.go

```diff
@@ -66,21 +67,32 @@ func (v *ChannelValidator) Handle(ctx context.Context, req admission.Request) ad
 		return admission.Allowed("")
 	}
 
-	if v, ok := isAllGit(chList); !ok {
+	inKey := types.NamespacedName{Name: chn.GetName(), Namespace: chn.GetNamespace()}
+	if v, ok := isGitOrSameKey(chList, inKey); !ok {
 		return admission.Denied(fmt.Sprintf("the namespace %v already contains a channel: %v",
 			chn.GetNamespace(), v))
 	}
 
 	return admission.Allowed("")
 }
 
-func isAllGit(chList *chv1.ChannelList) (string, bool) {
+//isGitOrSameKey will check: 1, if the exist channel in the request namespace
+// is a git git channel, if so, pass the request
+// 2. if the exist channel has the same name as the request channel, if so,
+// pass the request and let the k8s api server handle it(eventually reject
+//the request with 409)
+// 3. request all the requesting channel, that doesnt meet the above rule
+func isGitOrSameKey(chList *chv1.ChannelList, inReq types.NamespacedName) (string, bool) {
 	for _, chn := range chList.Items {
 		chnType := string(chn.Spec.Type)
 		if strings.EqualFold(chnType, chv1.ChannelTypeGit) || strings.EqualFold(chnType, chv1.ChannelTypeGitHub) {
 			continue
 		}
 
+		if chn.GetName() == inReq.Name {
+			return "", true
+		}
+
 		return chn.GetName(), false
 	}
```

Comment:
Hey @ianzhang366, thanks for fixing this. I think this will work. This code seems to depend on the order in which the channels are iterated, however, for non-Git channels, there should only be 1 channel in this loop, so I guess it won't actually cause a problem.